library/src/main/java/org/apache/apex/malhar/lib/window/TriggerOption.java

```diff
@@ -0,0 +1,278 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.window;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.joda.time.Duration;
+
+import org.apache.hadoop.classification.InterfaceStability;
+
+import com.esotericsoftware.kryo.serializers.FieldSerializer;
+import com.esotericsoftware.kryo.serializers.JavaSerializer;
+
+/**
+ * This class describes how triggers should be fired for each window.
+ * For each window, a trigger can be fired before the watermark (EARLY), at the watermark (ON_TIME), or after the watermark (LATE).
+ * If a LATE trigger is specified and the accumulation mode is ACCUMULATING, it is important for the WindowOption to
+ * specify the allowed lateness because otherwise, all states must be kept in storage.
+ *
+ */
+@InterfaceStability.Evolving
+public class TriggerOption
+{
+
+  public enum AccumulationMode
+  {
+    DISCARDING,
+    ACCUMULATING,
+    ACCUMULATING_AND_RETRACTING
+  }
+
+  private AccumulationMode accumulationMode = AccumulationMode.DISCARDING;
+  private boolean firingOnlyUpdatedPanes = false;
+
+  /**
+   * Whether the trigger should be fired before the watermark, at the watermark, or after the watermark
+   */
+  public enum WatermarkOpt
+  {
+    EARLY,
```

Comment:
If I understand correctly from bream model, EARLY, ON-TIME & LATE are the attributes of data and/or of the results to be send to the downstream.

Its not the attribute of a trigger. A trigger is an action which sends updated to downstream. The data containing that should be marked late, early or on-time.

Shouldn't this enum be moved out as its not specific to trigger?

Also the terminology of WatermarkOpt is confusion. As I said before these are not the attribute of watermark or trigger.